---
 ToolChains.cpp |    8 ++++++++
 1 file changed, 8 insertions(+)

Description: Add the multi-arch lib directory to the file search path
 Otherwise Clang is not able to find crt*.o.

Author: Michael Wild <themiwi@users.sourceforge.net>
Bug: http://llvm.org/bugs/show_bug.cgi?id=10100
Bug-Debian: http://bugs.debian.org/629594
Forwarded: http://llvm.org/bugs/attachment.cgi?id=6711
Last-Update: 2011-06-08

--- clang-2.9.orig/tools/clang/lib/Driver/ToolChains.cpp
+++ clang-2.9/tools/clang/lib/Driver/ToolChains.cpp
@@ -1531,6 +1531,11 @@ Linux::Linux(const HostInfo &Host, const
   Paths.push_back(Base + "/../../..");
   if (Arch == getArch() && IsUbuntu(Distro))
     Paths.push_back("/usr/lib/" + GccTriple);
+
+    if (!llvm::sys::fs::exists("/usr/lib/" + GccTriple + "/crt1.o", Exists) && Exists)
+      Paths.push_back("/usr/lib/" + GccTriple);
+    else if (!llvm::sys::fs::exists("/usr/lib/i386-linux-gnu/crt1.o", Exists) && Exists)
+      Paths.push_back("/usr/lib/i386-linux-gnu");
 }
 
 bool Linux::HasNativeLLVMSupport() const {
